NuRadioMC/utilities/Veff.py

```diff
@@ -375,6 +377,28 @@ def get_Veff(folder, trigger_combinations={}, station=101):
         V = area * dZ
         Vrms = fin.attrs['Vrms']
 
+        if(correct_zenith_sampling):
+            if(len(weights) > 0):
+                # correct weights to account for incorrect zenith sampling
+                from NuRadioMC.EvtGen.generator import draw_zeniths
+                zeniths1 = draw_zeniths(1e6, rmax, fin.attrs['zmax'], fin.attrs['zmin'], thetamin, thetamax)
+                u = np.random.uniform(np.cos(thetamax), np.cos(thetamin), int(1e6))
+                zeniths2 = np.arccos(u)
+                bins = np.linspace(thetamin, thetamax, 100)
+                N1, bin_edges = np.histogram(zeniths1, bins=bins)
+                N2, bin_edges = np.histogram(zeniths2, bins=bins)
+                X = (bins[1:] + bins[:-1]) * 0.5
+
+                def get_weights(zeniths):
+                    weights = np.zeros_like(zeniths)
+                    for i, z in enumerate(zeniths):
+                        mask = np.argmin(np.abs(z - X))
+                        weights[i] = (N2 / N1)[mask]
```

Comment:
I think you should use N1/N2 here, or directly use the normalised projected area.

Reply:
yes I think your right: For Moores Bay the projected area at the horizon is smaller than for the zenith. So with a ~sin(theta) distribution I have too many events at the horizon. Which means I need to weight them down.